src/test/app/AccountDelete_test.cpp

```diff
@@ -912,6 +912,301 @@ class AccountDelete_test : public beast::unit_test::suite
+                ter(tecBAD_CREDENTIALS));
+            env.close();
+
+            // becky accept the credentials
+            env(credentials::accept(becky, carol, credType));
+            env.close();
+
+            // Fail, credentials doesn’t belong to carol
+            env(acctdelete(carol, alice),
+                credentials::ids({credIdx}),
+                fee(acctDelFee),
+                ter(tecBAD_CREDENTIALS));
+
+            // Fail, no depositPreauth for provided credentials
+            env(acctdelete(becky, alice),
+                credentials::ids({credIdx}),
+                fee(acctDelFee),
+                ter(tecNO_PERMISSION));
+            env.close();
+
+            // alice create DepositPreauth Object
+            env(deposit::authCredentials(alice, {{carol, credType}}));
+            env.close();
+
+            // becky attempts to delete her account, but alice won't take her
+            // XRP, so the delete is blocked.
+            env(acctdelete(becky, alice),
+                fee(acctDelFee),
+                ter(tecNO_PERMISSION));
+
+            // becky use empty credentials and can't delete account
+            env(acctdelete(becky, alice),
+                fee(acctDelFee),
+                credentials::ids({}),
+                ter(temMALFORMED));
+
+            // becky use bad credentials and can't delete account
+            env(acctdelete(becky, alice),
+                credentials::ids(
+                    {"48004829F915654A81B11C4AB8218D96FED67F209B58328A72314FB6E"
+                     "A288BE4"}),
+                fee(acctDelFee),
+                ter(tecBAD_CREDENTIALS));
+            env.close();
+
+            // becky use credentials and can delete account
+            env(acctdelete(becky, alice),
+                credentials::ids({credIdx}),
+                fee(acctDelFee));
+            env.close();
+
+            // carol issue credentials for daria
+            env(credentials::create(daria, carol, credType));
+            env.close();
+            env(credentials::accept(daria, carol, credType));
+            env.close();
+            std::string const credDaria =
+                credentials::ledgerEntry(
+                    env, daria, carol, credType)[jss::result][jss::index]
+                    .asString();
+
+            // daria use valide credentials, which aren't required and can
+            // delete account
+            env(acctdelete(daria, carol),
+                credentials::ids({credDaria}),
+                fee(acctDelFee));
+            env.close();
+
+            // check that credential object deleted too
+            auto const jNoCred =
+                credentials::ledgerEntry(env, becky, carol, credType);
+
+            BEAST_EXPECT(
+                jNoCred.isObject() && jNoCred.isMember(jss::result) &&
+                jNoCred[jss::result].isMember(jss::error) &&
+                jNoCred[jss::result][jss::error] == "entryNotFound");
```

Comment:
This check should be before the Daria test.

Reply:
Please explain, which check and why? Check that object didn't exists before we delete it?

Reply:
This is checking the Becky/Carol credential, not the Daria credential. So it's confusing that this check is happening with another test in the middle.

Reply:
fixed